client/reader/user-profile/components/user-profile-header/index.tsx

```diff
@@ -18,17 +21,20 @@ interface UserProfileHeaderProps {
 const UserProfileHeader = ( { user }: UserProfileHeaderProps ): JSX.Element => {
 	const translate = useTranslate();
 	const currentPath = page.current;
-	const userProfileUrl = getUserProfileUrl( user.user_login ?? '' );
+	const userProfileUrlWithUsername = getUserProfileUrlWithUsername( user.user_login ?? '' );
+	const userProfileUrlWithId = getUserProfileUrlWithId( user.ID.toString() );
```

Comment:
Building off the first comment, we could just continue to use the original `getUserProfileUrl` function for getting both of these variables.